ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -511,13 +511,17 @@ public void compile(String command, boolean resetTaskIds, boolean deferClose) th
   }
 
   private void prepareForCompile(boolean resetTaskIds) throws CommandProcessorException {
-    driverTxnHandler.createTxnManager();
-    DriverState.setDriverState(driverState);
-    prepareContext();
-    setQueryId();
+    try {
```

Comment:
I suppose that this try-catch block was introduced to address the [so-called regression](https://github.com/apache/hive/pull/3393#issuecomment-1207815099) from HIVE-22526 but I don't fully understand where the regression comes from.

I had a quick look in the diff introduced by HIVE-22526 but as far as I can see there the`driverTxnHandler#createTxnManager` is not wrapped in a try-catch block and it seems that the exception is not passed in `handleException` method. @zhangbutao can you explain a bit more on where exactly is the regression?

Reply:
@zabetak Maybe I didn't describe it exactly. In fact, I tried to fix two related issues in this PR.

1. First: Beeline client can not get real exception message which is caused by the change(https://issues.apache.org/jira/browse/HIVE-23124).

2. Second: Even though the first issus is fixed, Beeline client can not get valid exception state and code in the following test:
`set hive.support.concurrency=false;`
`set hive.txn.manager=org.apache.hadoop.hive.ql.lockmgr.DbTxnManager;`
`create table testacid(id int) stored as orc tblproperties('transactional'='true');`
_**Error: Error running query: java.lang.RuntimeException: To use DbTxnManager you must set hive.support.concurrency=true (state=,code=0)**_
However, In hive3, we can get valiad exception with state=42000, code=10264
**_Error: Error while compiling statement: FAILED: RuntimeException [Error 10264]: To use DbTxnManager you must set hive.support.concurrency=true (state=42000,code=10264)_**
This change was introduced by https://issues.apache.org/jira/browse/HIVE-22526, which missed handling the exception from `driverTxnHandler#createTxnManager`, i also think we should fix this.